**Устройство состоит из трех деталей. Для первой детали вероятность
выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей -
0.25. Какова вероятность того, что в первый месяц выйдут из строя:**  

                а) все детали; 
                б) только две детали;  
                в) хотя бы одна деталь;  
                г) от одной до двух деталей?

In [1]:
import numpy as np
from math import factorial

Введем следующие события:  
а) Событие $A_1$-вышла из строя первая деталь, по условию
задачи-вероятность этого события: $P(A_1)=0.1$;  
б) Событие $\bar A_1$-противоположное событию $A_1$ - первая деталь не
вышла из строя, вероятность этого события:
$P(\bar A_1)=1-P(A_1)=1-0.1=0,9$;  
в) Событие $A_2$-вышла из строя вторая деталь, по условию
задачи-вероятность этого события: $P(A_2)=0.2$;  
г) Событие $\bar A_2$-противоположное событию $A_2$ - вторая деталь не
вышла из строя, вероятность этого события:
$P(\bar A_2)=1-P(A_2)=1-0.2=0,8$;  
д) Событие $A_3$-вышла из строя первая деталь, по условию
задачи-вероятность этого события: $P(A_3)=0.25$;  
е) Событие $\bar A_3$-противоположное событию $A_3$ - третья деталь не
вышла из строя, вероятность этого события:
$P(\bar A_3)=1-P(A_3)=1-0.25=0,9$;

------------------------------------------------------------------------

а) Для того, чтобы произошло событие "вышли из строя все детали",
необходимо, чтобы **одновременно** произошли события $A_1$, $A_2$, и
$A_3$. А вероятностью этого события $P(3)$ будет произведением
вероятностей этих событий.  
$P(3)=P(A_1)\cdot P(A_2)\cdot P(A_3)=0.1\cdot 0.2\cdot 0.25$

In [2]:
P3=0.1*0.2*0.25
print(f'Вероятность того, что из строя выйдут все детали Р(3) = {P3: .4f}')

Вероятность того, что из строя выйдут все детали Р(3) =  0.0050


------------------------------------------------------------------------

б) Для того, чтобы произошло событие "вышли из строя 2 детали",
необходимо что бы случилось **одна из** следующих комбинаций событий:

1.  Произошли события $A_1$ **и** $A_2$ **и** $\bar A_3$;  
2.  Произошли события $A_1$ **и** $A_3$ **и** $\bar A_2$;  
3.  Произошли события $A_2$ **и** $A_3$ **и** $\bar A_1$.

А вероятностью этого события $P(2)$ будет **суммой** вероятностей этих
комбинаций событий.  
$P(2)=P(A_1)\cdot P(A_2)\cdot P(\bar A_3)+P(A_1)\cdot P(A_3)\cdot P(\bar A_2)+P(A_2)\cdot P(A_3)\cdot P(\bar A_1)=0.1\cdot 0.2\cdot 0.75+0.1\cdot 0.25\cdot 0.8+0.2\cdot 0.25\cdot 0.9$

In [3]:
P2=0.1*0.2*0.75+0.1*0.25*0.8+0.2*0.25*0.9
print(f'Вероятность того, что из строя выйдут только 2 детали Р(2) = {P2: .4f}')

Вероятность того, что из строя выйдут только 2 детали Р(2) =  0.0800


------------------------------------------------------------------------

в) Событие "выйдет из строя хотя бы одна деталь" можно рассмотреть как
противоположное событию "не вышло из строя ни одной детали".  
Пусть $P(0)$ вероятность события "не вышло из строя ни одной детали".
Данное событие может произойти при одновременном наступлении событий
$\bar A_1$, $\bar A_2$, и $\bar A_3$.  
$P(0)=P(\bar A_1)\cdot P(\bar A_2)\cdot P(\bar A_3)=0.9*0.8*0.75$

In [ ]:
P0=0.9*0.8*0.75
print(f'Вероятность того, что из строя не выйдет ни одной детали Р(0) = {P0: .4f}')

Вероятность того, что из строя не выйдет ни одной детали Р(0) =  0.5400


Тогда вероятность события "выйдет из строя хотя бы одна деталь"
$P(\ge 1)=1-P(0)$

In [ ]:
P_0=1-P0
print(f'Вероятность того, что выйдет из строя хотя бы одна деталь Р(>=1) = {P_0: .4f}')

Вероятность того, что выйдет из строя хотя бы одна деталь Р(>=1) =  0.4600


г) Событие "из строя выйдут от одной до двух деталей" произойдет при
наступлении события "из строя выйдет одна деталь" **или** из строя
выйдут 2 детали, следовательно его вероятность будет равна сумме
вероятностей этих событий: $P(1,2)=P(1)+P(2)$  
$P(2)=0.08$, найдем $P(1)$:

<u>**Способ 1:**</u>

Для того, чтобы произошло событие "из строя выйдет одна деталь",
необходимо что бы случилось **одна из** следующих комбинаций событий:

1.  Произошли события $A_1$ **и** $\bar A_2$ **и** $\bar A_3$;  
2.  Произошли события $A_2$ **и** $\bar A_1$ **и** $\bar A_3$;  
3.  Произошли события $A_3$ **и** $\bar A_1$ **и** $\bar A_2$.



А вероятностью этого события $P(1)$ будет **суммой** вероятностей этих
комбинаций событий.  
$P(1)=P(A_1)\cdot P(\bar A_2)\cdot P(\bar A_3)+P(A_2)\cdot P(\bar A_1)\cdot P(\bar A_3)+P(A_3)\cdot P(\bar A_1)\cdot P(\bar A_2)=0.1\cdot 0.8\cdot 0.75+0.2\cdot 0.9\cdot 0.75+0.25\cdot 0.9\cdot 0.8$  

In [ ]:
P1=0.1*0.8*0.75+0.2*0.9*0.75+0.25*0.9*0.8
print(f'Вероятность того, что выйдет из строя одна деталь Р(1) = {P1: .4f}')

Вероятность того, что выйдет из строя одна деталь Р(1) =  0.3750


<u>**Способ 2:**</u>

События "не выйдет из строя ни одной детали", "из строя выйдет одна
деталь", "из строя выйдет 2 детали", "из строя выйдут все детали" -
**несовместны**, и образуют полную группу событий, поэтому сумма
вероятностей наступления этих событий равна $1$.  
$P(0)+P(1)+P(2)+P(3)=1\\0.54+P(1)+0.08+0.005=1\\P(1)=0.375$  


$P(1,2)=P(1)+P(2)=0.375+0.08=0.455$

------------------------------------------------------------------------

## Ответ:

а) Вероятность того что в первый месяц выйдут из строя все детали
$P(3)=0.005$ или $0.50\%$  
б) Вероятность того что в первый месяц выйдут из строя 2 детали
$P(2)=0.08$ или $8.00\%$  
в) Вероятность того что в первый месяц выйдут из строя хотя бы одна
деталь $P(\ge 1)=0.46$ или $46.00\%$  
г) Вероятность того что в первый месяц выйдут из строя от одной до двух
деталей $P(1,2)=0.455$ или $45.50\%$

------------------------------------------------------------------------